src/features/collection/Collection.tsx

```diff
@@ -90,7 +92,13 @@ const Collection: React.FC<{}> = () => {
 
             <div className='w-1/2 flex items-center justify-end'>
               <SearchBox onChange={handleSearchChange} placeholder='Filter' dark />
-              <Link to='/ds/new'>
+              <Link to={{
+                pathname: `/ds/${username}/dataset_${Math.floor(Math.random() * 1000)}/workflow/new`,
```

Comment:
we can tackle this in a follow up, but we should drop the random dataset name generation